src/dstore/pmix_esh.c

```diff
@@ -1056,18 +1085,18 @@ int _esh_fetch(const char *nspace, pmix_rank_t rank, const char *key, pmix_value
              * EXTENSION slot which has key = EXTENSION_SLOT and a size_t value for offset
              * to next data address for this process.
              */
-            if (0 == strncmp((const char *)addr, ESH_REGION_INVALIDATED, strlen(ESH_REGION_INVALIDATED)+1)) {
+            if (0 == strncmp((const char *)addr + sizeof(size_t), ESH_REGION_INVALIDATED, KNAME_LEN(ESH_REGION_INVALIDATED))) {
```

Comment:
Let's do a macro for each DSTORE field access to avoid explicit `addr + sizeof(x)` things.